core/autoscaling.go

```diff
@@ -536,6 +596,19 @@ func (a *autoScalingGroup) getAllowedInstanceTypes(baseInstance *instance) []str
 
 }
 
+func (a *autoScalingGroup) getPricetoBid(
+	baseOnDemandPrice float64, currentSpotPrice float64) float64 {
+	logger.Println("a.BiddingPolicy: ", a.BiddingPolicy)
+	if a.BiddingPolicy == "aggressive" {
+		logger.Println("Launching spot instance with a bid =", currentSpotPrice*a.region.conf.SpotPriceMultiplier)
+		return currentSpotPrice * a.region.conf.SpotPriceMultiplier
```

Comment:
I realised that there might be a potential issue here, if the spot price is close to the on-demand price for some reasons (or the multiplier very high), then having the SpotPriceMultiplier might allow the bidding to go above that value. e.g.:

```bash
spotPrice: 0.5
onDemandPrice: 1
spotPriceMultiplier: 3.0
spotBid: => 1.5

# Other case
spotPrice: 0.9 (couldn't find better)
onDemandPrice: 1
spotPriceMultiplier: 1.5
spotBid: => 1.35
```

I think we might want to check that this potential maximum price isn't higher than the on-demand price, just for the sake of avoiding it; or we might end with spot instances running costing more than on-demand.

Reply:
Good catch!

We can just take the minimum between ondemand price and the calculated spot price

Reply:
Cool, will do.